### PR TITLE
Added Rust 1.59.0

### PR DESCRIPTION
**Pull request by UebelAndre**
https://blog.rust-lang.org/2022/02/24/Rust-1.59.0.html

### CODE REVIEW

File: tools/rustfmt/srcs/main.rs [hunk 1]
```diff
@@ -109,7 +109,7 @@ fn apply_rustfmt(options: &Config, targets: &[String]) {
 
     for target in targets.iter() {
         // Replace any `:` characters and strip leading slashes
-        let target_path = target.replace(":", "/").trim_start_matches('/').to_owned();
+        let target_path = target.replace(':', "/").trim_start_matches('/').to_owned();
```
Comment (hlopko) [inline tools/rustfmt/srcs/main.rs:112]: Why these changes??

Comment (UebelAndre) [reply tools/rustfmt/srcs/main.rs:112]: https://github.com/bazelbuild/rules_rust/pull/1153/commits/4ef1f96b94f10131eac25f18d03630683485bb7b - New clippy defects.

Comment (hlopko) [reply tools/rustfmt/srcs/main.rs:112]: Aaah I see, thank you!

**Review by hlopko — APPROVED** [review]
